src/citrine/resources/team.py

```diff
@@ -0,0 +1,200 @@
+"""Resources that represent both individual and collections of teams."""
+from typing import Optional, Union, List
+from uuid import UUID
+
+from citrine._rest.collection import Collection
+from citrine._rest.resource import Resource, ResourceTypeEnum
+from citrine._serialization import properties
+from citrine._session import Session
+from citrine._utils.functions import format_escaped_url
+
+WRITE = "WRITE"
+READ = "READ"
+SHARE = "SHARE"
+ACTIONS = Union[WRITE, READ, SHARE]
+
+
+class TeamMember:
+    """A Member of a Team."""
+
+    def __init__(self,
+                 *,
+                 id: UUID,
+                 screen_name: str,
+                 email: str,
+                 is_admin: bool,
+                 team: 'Team',  # noqa: F821
+                 actions: ACTIONS):
+        self.id = id
+        self.screen_name = screen_name
+        self.email = email
+        self.is_admin = is_admin
+        self.team: 'Team' = team  # noqa: F821
+        self.actions: ACTIONS = actions
+
+    def __str__(self):
+        return f'<TeamMember {self.screen_name} is MEMBER of {self.team.name}>'
+
+
+class Team(Resource['Team']):
+    """
+    A Citrine Team.
+
+    A Team is a collection of projects and resources.
+
+    Parameters
+    ----------
+    name: str
+        Name of the team.
+    description: str
+        Long-form description of the team.
+    session: Session, optional
+        The Citrine session used to connect to the database.
+
+    Attributes
+    ----------
+    uid: UUID
+        Unique uuid4 identifier of this team.
+    created_at: int
+        Time the team was created, in seconds since epoch.
+
+    """
+
+    _response_key = 'team'
+    _resource_type = ResourceTypeEnum.TEAM
+    _api_version = "v3"
+
+    name = properties.String('name')
+    description = properties.Optional(properties.String(), 'description')
+    uid = properties.Optional(properties.UUID(), 'id')
+    created_at = properties.Optional(properties.Datetime(), 'created_at')
+
+    def __init__(self,
+                 name: str,
+                 *,
+                 description: str = "",
+                 session: Optional[Session] = None):
+        self.name: str = name
+        self.description: str = description
+        self.session: Session = session
+
+    def __str__(self):
+        return '<Team {!r}>'.format(self.name)
+
+    def _path(self):
+        return format_escaped_url('/teams/{team_id}', team_id=self.uid)
+
+    def list_members(self) -> List[TeamMember]:
+        """List the Team Members."""
+        members = self.session.get_resource(self._path() + "/users",
+                                            version=self._api_version)["users"]
+        return [TeamMember(id=m["id"], screen_name=m["screen_name"], email=m["email"],
```

Comment:
instead of a new TeamMembers object, see if we can make Users backwards compatible?